### PR TITLE
Adding automate backend elasticsidecar

### PR DESCRIPTION
**Pull request by FaizanSRE**
### :nut_and_bolt: Description: What code changed, and why?
For Code Movement to the Automate Repo we have 12 components, so as part of this PR we have targeting below components.
automate backend elasticsidecar

### :+1: Definition of Done
As Part of this we want the components to move in automate-repo and can be build their habitat package in that environment.
### :athletic_shoe: How to Build and Test the Change
To build this package

Cloning this PR changes
Enter hab studio using  -- > hab studio enter
build respective components in hab studio like for example , build components/automate-backend-elasticsidecar

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb [hunk 1]
```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env ruby
+
+require 'bcrypt'
```
Comment (tas50) [inline components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb:3]: Chefstyle should be applied to this code

File: components/automate-backend-elasticsidecar/habitat/plan.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+pkg_description="Password manager for elasticsearch-odfe"
+pkg_origin="chef"
+vendor_origin="chef"
+pkg_maintainer="Chef Software Inc. <support@chef.io>"
+pkg_license=("Chef-MLSA")
+pkg_version="1.0.28"
+pkg_upstream_url="http://github.com/chef/a2-ha-backend/components/automate-backend-elasticsidecar"
+pkg_deps=(
+  core/bash
+  core/ruby26
```
Comment (tas50) [inline components/automate-backend-elasticsidecar/habitat/plan.sh:12]: Is there a reason we're building tools on a version of Ruby that's going to be EOL at the end of the year?

File: components/automate-backend-elasticsidecar/habitat/plan.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+do_before() {
+  update_pkg_version
+}
+
+do_build() {
+  return 0
+}
+
+do_install() {
+  gem update --system
```
Comment (tas50) [inline components/automate-backend-elasticsidecar/habitat/plan.sh:38]: There's no real benefit in doing this with modern ruby and it just makes the packages bigger

**Comment by tas50** [comment]
In general this would all work a lot safer if we did this with a proper gemspec and a bundled gemfile.lock. As it stands right now the dependencies can change from build to build, which is a big problem since we lack reproducible builds with that approach.

File: components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  spec.version       = Automate::Backend::Elasticsidecar::VERSION
+  spec.authors       = ["Faizan Fulara"]
+  spec.email         = ["ffulara@progress.com"]
+  spec.description   = %q{Package to control backend-elasticsidecar}
+  spec.summary       = spec.description
+  spec.licenses      = "Apache-2.0"
+
+  spec.require_paths = ["lib"]
+
+  spec.required_ruby_version = ">= 2.4.0"
```
Comment (tas50) [inline components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec:16]: This is an EOL ruby release. We shouldn't support it.

File: components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  spec.add_development_dependency "bundler"
+   
+  spec.add_runtime_dependency 'bcrypt'
+  spec.add_runtime_dependency 'http'
+  spec.add_runtime_dependency 'json'
```
Comment (tas50) [inline components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec:21]: This is built in

File: components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  spec.authors       = ["Faizan Fulara"]
+  spec.email         = ["ffulara@progress.com"]
+  spec.description   = %q{Package to control backend-elasticsidecar}
+  spec.summary       = spec.description
+  spec.licenses      = "Apache-2.0"
+
+  spec.require_paths = ["lib"]
+
+  spec.required_ruby_version = ">= 2.4.0"
+  spec.add_development_dependency "bundler"
```
Comment (tas50) [inline components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec:17]: This is built in to modern ruby

File: components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+require 'fileutils'
+require 'http'
+require 'json'
+require 'logger'
+require 'toml-rb'
+require 'yaml'
+require 'mixlib/shellout'
+require "chefstyle"
```
Comment (tas50) [inline components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb:11]: ?

File: components/automate-backend-elasticsidecar/habitat/plan.sh (hunk 2 above)
Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/plan.sh:12]: We Will be upgrading this in subsequent sprints https://app.zenhub.com/workspaces/the-delta-quadrant-5fc7c4c3922dc10021323fa6/issues/chef/automate/5546

File: components/automate-backend-elasticsidecar/habitat/plan.sh (hunk 3 above)
Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/plan.sh]: Okay Noted, Have removed this

File: components/automate-backend-elasticsidecar/habitat/plan.sh [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  gem install bcrypt --no-document --install-dir "${pkg_prefix}/lib/gems"
```
[the remark this answers is not in the document]
Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/plan.sh]: Created the gemspec file and pulling gems there

File: components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec (hunk 4 above)
Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec:16]: Okay , will upgrade the support to 2.6 and greater

File: components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec (hunk 5 above)
Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec:21]: Okay will update accordingly

File: components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec (hunk 6 above)
Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/automate-backend-elasticsidecar.gemspec:17]: Okay will update accordingly

File: components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb (hunk 7 above)
Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/bin/elastic_sidecar.rb:11]: So how we can use this chef-style , I was trying to use this for chef-style lintment

File: components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/CODE_OF_CONDUCT.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
+# Contributor Covenant Code of Conduct
```
Comment (kalroy) [inline components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/CODE_OF_CONDUCT.md:1]: There is no need of this file. Automate repo root already has this file

File: components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,5 @@
+## [Unreleased]
```
Comment (kalroy) [inline components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/CHANGELOG.md:1]: We do not maintain Changelog in automate

**Review by kalroy — CHANGES_REQUESTED** [review]
The description does not have enough information to run the service in hab studio

File: components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/CODE_OF_CONDUCT.md (hunk 9 above)
Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/CODE_OF_CONDUCT.md:1]: ok removing it

File: components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/CHANGELOG.md (hunk 10 above)
Comment (FaizanSRE) [reply components/automate-backend-elasticsidecar/habitat/automate-backend-elasticsidecar/CHANGELOG.md:1]: okay removing it

**Review by vivek-yadav — APPROVED** [review]

**Pull request merged**